potential compile error with unity build and deal.II 9.2.0

When changing the number of batched files with unity build, I am running into a deal.II bug that has been fixed after 9.2:
```
/Unity/unity_2_cxx.cxx.o.d -o CMakeFiles/aspect.dir/Unity/unity_2_cxx.cxx.o -c CMakeFiles/aspect.dir/Unity/unity_2_cxx.cxx
In file included from /hdscratch/shared/libs-candi-9.2.0-r1/deal.II-v9.2.0/include/deal.II/fe/mapping_cartesian.h:22:0,
                 from /home/heister/aspect-git/source/simulator/core.cc:53,
                 from CMakeFiles/aspect.dir/Unity/unity_2_cxx.cxx:97:
/hdscratch/shared/libs-candi-9.2.0-r1/deal.II-v9.2.0/include/deal.II/base/qprojector.h:414:54: error: specialization of ‘dealii::QIterated<dim>::QIterated(const dealii::Quadrature<1>&, unsigned int) [with int dim = 1]’ after instantiation
                         const unsigned int   n_copies);
```
see https://github.com/dealii/dealii/commit/1e5d09f19a26eb85cbf3a360586592283e358c90#diff-cababb69c891317230d592c968e1c64f for the bugfix.
We will compile cleanly as long as no header that includes qiterated.h is batched together with core.cc. This is not the case with default settings, but it might change randomly if a user adds another .cc file.
What to do: exclude core.cc from unity build (for now)?
Ignore this until it breaks something?
Any other suggestions? (include qprojector in every file before qiterated?)

@bangerth do you have an idea?

> Can you show the complete error message?

This is the complete error message
```
[3+2/5 71.015s] Building CXX object CMakeFiles/aspect.dir/Unity/unity_2_cxx.cxx.o
FAILED: CMakeFiles/aspect.dir/Unity/unity_2_cxx.cxx.o
/usr/bin/mpicxx -DBOOST_NO_AUTO_PTR -DDEBUG -DTBB_DO_ASSERT=1 -DTBB_USE_DEBUG -I../include -Iinclude -I../contrib/catch -isystem /hdscratch/
shared/libs-candi-9.2.0-r1/deal.II-v9.2.0/include -isystem /hdscratch/shared/libs-candi-9.2.0-r1/deal.II-v9.2.0/include/deal.II/bundled -isy
stem /hdscratch/shared/libs-candi-9.2.0-r1/parmetis-4.0.3/include -isystem /hdscratch/shared/libs-candi-9.2.0-r1/trilinos-release-12-18-1/in
clude -isystem /usr/include/suitesparse -isystem /hdscratch/shared/libs-candi-9.2.0-r1/petsc-3.13.1/include -isystem /hdscratch/shared/libs-
candi-9.2.0-r1/hdf5-1.10.5/include -isystem /hdscratch/shared/libs-candi-9.2.0-r1/oce-OCE-0.18.2/include/oce -isystem /hdscratch/shared/libs
-candi-9.2.0-r1/p4est-2.2/FAST/include -isystem /hdscratch/shared/libs-candi-9.2.0-r1/slepc-3.13.2/include -pedantic -fPIC -Wall -Wextra -Wo
verloaded-virtual -Wpointer-arith -Wsign-compare -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-literal-suffix -Wno-psabi -fopenmp
-simd -std=c++17 -Wno-parentheses -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections -Winvalid-pch -include /home/heister/as
pect-git/build-unitytest/CMakeFiles/aspect.dir/cmake_pch.hxx -MD -MT CMakeFiles/aspect.dir/Unity/unity_2_cxx.cxx.o -MF CMakeFiles/aspect.dir
/Unity/unity_2_cxx.cxx.o.d -o CMakeFiles/aspect.dir/Unity/unity_2_cxx.cxx.o -c CMakeFiles/aspect.dir/Unity/unity_2_cxx.cxx
In file included from /hdscratch/shared/libs-candi-9.2.0-r1/deal.II-v9.2.0/include/deal.II/fe/mapping_cartesian.h:22:0,
                 from /home/heister/aspect-git/source/simulator/core.cc:53,                               
                 from CMakeFiles/aspect.dir/Unity/unity_2_cxx.cxx:97:
/hdscratch/shared/libs-candi-9.2.0-r1/deal.II-v9.2.0/include/deal.II/base/qprojector.h:414:54: error: specialization of ‘dealii::QIterated<d
im>::QIterated(const dealii::Quadrature<1>&, unsigned int) [with int dim = 1]’ after instantiation
                         const unsigned int   n_copies);                                                  
                                                      ^        
[1+4/5 82.859s] Building CXX object CMakeFiles/aspect.dir/Unity/unity_1_cxx.cxx.o
ninja: build stopped: subcommand failed.               
```

> * Just write the missing declaration into our own files.

It is not a "missing" declaration, but a nonsensical specialization, so I don't see how this can work. One would need to include qprojector.h before quadrature.h. I will prepare a patch.

--- a/include/aspect/material_model/interface.h
+++ b/include/aspect/material_model/interface.h
@@ -26,7 +26,17 @@
 #include <aspect/material_model/utilities.h>
 
 #include <deal.II/base/point.h>
+
+// Work around an incorrect instantiation in qprojector.h of deal.II 9.2.0,
+// which requires including qprojector.h before quadrature.h (and not
+// after). This file doesn't actually need qprojector.h, so the include can be
+// removed when we require 9.3.. For more info see
+// https://github.com/geodynamics/aspect/issues/3728
+#if !DEAL_II_VERSION_GTE(9,3,0)
+#include <deal.II/base/qprojector.h>
+#endif
 #include <deal.II/base/quadrature.h>
+
 #include <deal.II/base/symmetric_tensor.h>
 #include <deal.II/base/parameter_handler.h>
 #include <deal.II/dofs/dof_handler.h>

